### PR TITLE
Adds salvage magnet board to the circuit imprinter

### LINKED ISSUE #31994: No Way To Get A New Salvage Magnet IC
## Description

Whenever a meteor blows it up and the board's gone, they'd need an admin to spawn a new one. Maybe either make the board indestructible or salv can somehow buy/craft a new one that gets locked when one already exists?

### PR DESCRIPTION
**Pull request by ArtisticRoomba**
## About the PR

Adds the salvage magnet board to the circuit imprinter (available roundstart), and to QM's locker. 
## Why / Balance

The salvage magnet board is currently unable to be obtained IC. You either have to admeme it into existence or simply suffer, as the salvage gameplay loop literally cannot be started with the magnet missing.

If your salvage magnet gets emogarbaged (meteored), stolen, or otherwise obliterated it can now be easily repaired. This PR fixes #31994.

I don't think this should be under research, science partially relies on Cargo for their gameplay loop and it would just be a happier situation if this could be made roundstart, it's a pretty integral thing for salvage.

The recipe requires gold for the circuitboard but the RD can easily ransack the vault for some gold if necessary. If this needs to be changed (because RD _somehow_ spent all the station's gold on useless shit) let me know. Same with the technology thing.

_Reminder of the TODO: Make the salvage magnet require HV wire and draw a lot of power when active, to mechanically enforce having the magnet being the "starter" for salvage's gameloop. Would require map changes._

## Technical details

YAML editing lmao.
Adds `SalvageMagnetMachineCircuitboard` to electronics.yml
Adds `SalvageMagnetMachineCircuitboard` to lathe.yml under `staticRecipes` if you're having a _really_ bad day.
Adds the supply sprite component to the circuitboard.

## Media

Media shows changed sprite, components needed, the double salvage magnet situation staying consistent, and the flatpacked version of the magnet.

Content Client_ebl7ZHmT4s

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

**Changelog**

:cl:
- tweak: The salvage magnet circuitboard has been added to QM's locker. It can also be made at the circuit imprinter roundstart. NOW GET BACK TO WORK!!!

### CODE REVIEW

**Comment by Cojoke-dot** [comment]
Just add it to the qm's closet, else sci salv would be too prevalent. It would just replace the spare reclaimer shuttle board anyways.

**Comment by ArtisticRoomba** [comment]
Sure.

**Comment by lzk228** [comment]
to resolve the issue i would just made salv magnet not completely desctruclible but turned into "broken" version that can be repaired

**Comment by Cojoke-dot** [comment]
> to resolve the issue i would just made salv magnet not completely desctruclible but turned into "broken" version that can be repaired

meh, meteors would end up just knocking it off the grid into deep space then

**Comment by ArtisticRoomba** [comment]
yeah, meteors can just knock it out into deep space, and with the magnet being so integral to salv's gameloop, it would be better to have this be on a lathe + in locker

if we feel that it needs to be behind technology then I can add it to a salvage-related tech that makes sense

**Comment by lzk228** [comment]
i'm just afraid that salv will abuse two or even three magnets
or it's not possible?

**Comment by lzk228** [comment]
that's fine then

**Comment by ArtisticRoomba** [comment]
What

**Comment by ArtisticRoomba** [comment]
Oh, #31524 lmao
Now I don't have to worry about random mats pricing

File: Resources/Prototypes/Recipes/Lathes/electronics.yml [hunk 1]
```diff
@@ -983,3 +983,13 @@
   materials:
     Steel: 100
     Glass: 500
+
+- type: latheRecipe
+  id: SalvageMagnetMachineCircuitboard
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Lathes/electronics.yml:988]: no reason for this to take extra steel or be 10 seconds, just inherit the gold electronics recipe

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by ArtisticRoomba** [comment]
also @deltanedas related issue to your PR #31524 while we're here, the base protos for `BaseGoldCircuitboardRecipe` and `BaseCircuitboardRecipe` show despite `abstract: true` in the lathe
Content Client_CfAyLjFJid

**Comment by beck-thompson** [comment]
Just had a shift where a meteorite exploded our magnet... Not fun we just sat around for the rest of the shift!

**Comment by Emisse** [comment]
shure

**Pull request merged**